native/libp2p_port/internal/reqresp/reqresp.go

```diff
@@ -43,17 +43,24 @@ func NewListener(p *port.Port, config *proto_helpers.Config) Listener {
 	return Listener{hostHandle: h, port: p, pendingMessages: make(map[string]chan []byte)}
 }
 
+func (l *Listener) Host() host.Host {
+	return l.hostHandle
+}
+
 func (l *Listener) HostId() []byte {
 	return []byte(l.hostHandle.ID())
 }
 
 func (l *Listener) AddPeer(id []byte, addrs []string, ttl int64) {
+	peerID := peer.ID(id)
+	peerstore := l.hostHandle.Peerstore()
 	for _, addr := range addrs {
 		maddr, err := multiaddr.NewMultiaddr(addr)
 		// TODO: return error to caller
 		utils.PanicIfError(err)
-		l.hostHandle.Peerstore().AddAddr(peer.ID(id), maddr, time.Duration(ttl))
+		peerstore.AddAddr(peerID, maddr, time.Duration(ttl))
 	}
+	l.hostHandle.Connect(context.TODO(), peerstore.PeerInfo(peerID))
```

Comment:
Good catch! I did it this way because in the NIF we have both calls as independent functions, and here we merged them. I'll modify this to first construct the `AddrInfo` and then only `Connect`